AlphabetClassification/app.py

```diff
@@ -0,0 +1,106 @@
+from matplotlib import pyplot as plt
+import streamlit as st
+import pandas as pd
+from sklearn.model_selection import train_test_split
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import accuracy_score, classification_report
+from sklearn.metrics import confusion_matrix, ConfusionMatrixDisplay
+
+
+
+# Page Configuraion
+st.set_page_config(
+    page_title ='EEG Alphabet Predictor App',
+    layout = 'wide',
+    initial_sidebar_state = 'expanded',
+)
+
+# Title of the Application
+st.title('EEG Alphabet Predictor')
+
+# Load Dataset
+data_file = st.file_uploader("Upload Your CSV file",type=['csv'])
+
+if data_file:
+    df = pd.read_csv(data_file)
+    st.info("EEG Dataset Uploaded")
+
+else:
+    # Default Dataset if user did not upload any csv file.
+    df = pd.read_csv('train.csv')
+    st.info("Using default Dataset")
+
+# Dataset Display and filtering rows which have target values in {0,1,2}
+# corresponding to {'A','B'.'C'}.
+st.subheader("Dataset Preview")
+#filtered_dataset = df[df['# Letter'].isin([0,1,2])]
+filtered_dataset = df[df['# Letter'].isin(range(26))]
+st.write(filtered_dataset)
+
+# Features and target seperated
+st.sidebar.subheader('Input Features')
+feature_columns = [col for col in df.columns if col not in ['Line', '# Letter']]
+X = df[feature_columns]
+y = df['# Letter']
+
+# Model building.
+rf = RandomForestClassifier(max_depth = 6,max_features=5,n_estimators=200,random_state=42)
+x_train,x_test,y_train,y_test = train_test_split (X,y,test_size=0.2,random_state=42)
+rf.fit(x_train,y_train)
+
+
+# Selecting top 6 features.
+importance = rf.feature_importances_
+
+# Displaying the importance of each feature.
+feature_importance_df = pd.DataFrame({
+    'Feature':feature_columns,
+    'Importance':importance
+}).sort_values(by = 'Importance',ascending=False)
+top_features = feature_importance_df.head(10)
+top_features_names = top_features['Feature'].tolist()
+
+
+# Data Splitting.
+# Setting a random_state makes sure the data split is the same every time the code is running.
+x_train,x_test,y_train,y_test = train_test_split (X[top_features_names],y,test_size=0.3,random_state=45)
+
+# Train the model with only top features.
+rf.fit(x_train,y_train)
+
+st.subheader('Displaying importance of each feature')
+# Displaying importance of each feature in the form of bar chart.
+st.bar_chart(feature_importance_df.set_index('Feature'))
+
+# Input Widges which only consist of Top features as input.
+input_features = pd.DataFrame([{feature:st.sidebar.slider(
+    feature, float(df[feature].min()),float(df[feature].max()), float(df[feature].mean())) 
+    for feature in top_features_names}],columns=top_features_names)
+st.subheader('Input Features')
+st.write(input_features)
+
+
+# Prediction part
+number_to_alphabet = {i: chr(65+i) for i in range(26)}

```

Comment:
Actually the classification is  between "A" and "B" only

Reply:
Hey @SaranDharshanSP, I'll make the required changes. But I'm wondering if we should focus only on classifying "A" and "B" or include "C" as well. Also, after making the changes, should I create a new demo for this specific classification, or should we keep the previous one that classifies all the alphabets?

Reply:
> Hey @SaranDharshanSP, I'll make the required changes. But I'm wondering if we should focus only on classifying "A" and "B" or include "C" as well. Also, after making the changes, should I create a new demo for this specific classification, or should we keep the previous one that classifies all the alphabets?

A and B only as other issues too are trying on A/B only.